python/lsst/ap/association/association.py

```diff
@@ -405,6 +406,36 @@ def _check_dia_object_position(self, dia_object_record, bbox, wcs):
                                                 geom.degrees))
         return bbox.contains(point)
 
+    def check_dia_souce_radec(self, dia_sources):
+        """Check that all DiaSources have non-NaN values for RA/DEC.
+
+        If one or more DiaSources are found to have NaN values, throw a
+        warning to the log with the ids of the offending sources. Drop them
+        from the table.
+
+        Parameters
+        ----------
+        dia_sources : `pandas.DataFrame`
+            Input DiaSources to check for NaN values.
+
+        Returns
+        -------
+        trimmed_sources : `pandas.DataFrame`
+            DataFrame of DiaSources trimmed of all entries with NaN values for
+            RA/DEC.
+        """
+        nan_mask = np.logical_or(np.isnan(dia_sources.loc[:, "ra"]),
```

Comment:
I would do this in a more native pandas way:

`nan_mask = dia_sources["ra"].isnull() |  dia_sources["decl"].isnull()`